pkg/providers/tinkerbell/reconciler/reconciler_test.go

```diff
@@ -283,24 +360,26 @@ func dataCenter(opts ...datacenterOpt) *anywherev1.TinkerbellDatacenterConfig {
 	return d
 }
 
-type tinkerbellMachineOpt func(config *anywherev1.TinkerbellMachineConfig)
+type tinkerbellMachineOpt func(config *v1alpha1.TinkerbellMachineConfig)
 
-func machineConfig(opts ...tinkerbellMachineOpt) *anywherev1.TinkerbellMachineConfig {
-	m := &anywherev1.TinkerbellMachineConfig{
+func machineConfig(opts ...tinkerbellMachineOpt) *v1alpha1.TinkerbellMachineConfig {
+	m := &v1alpha1.TinkerbellMachineConfig{
 		TypeMeta: metav1.TypeMeta{
-			Kind:       anywherev1.TinkerbellMachineConfigKind,
-			APIVersion: anywherev1.GroupVersion.String(),
+			Kind:       v1alpha1.TinkerbellMachineConfigKind,
+			APIVersion: v1alpha1.GroupVersion.String(),
 		},
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: clusterNamespace,
 		},
-		Spec: anywherev1.TinkerbellMachineConfigSpec{
-			OSFamily:         "ubuntu",
-			HardwareSelector: anywherev1.HardwareSelector{},
-			Users: []anywherev1.UserConfiguration{
```

Comment:
Why change the test from using "ubuntu" to "bottlerocket"?

Reply:
Since we add all the new validations now, and ubuntu can't be auto-imported, we use bottlerocket to avoid the auto import error. We have to provide an additional OS Image URL for ubuntu.